.github/workflows/release.yml

```diff
@@ -0,0 +1,92 @@
+name: Release new library version
+
+# Controls when the workflow will run
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+
+  release-latest:
+
+    permissions:
+      id-token: write   # to verify the deployment originates from an appropriate source
+      contents: write   # To allow pushing tags/etc.
+
+    # Specify runner + deployment step
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: pip install .
+      - name: version-tag
+        # Grab the current version of the library and create a tag like `v{version}`
+        run: |
+          echo "$(python -m sync --version)" |
+          sed -nE 's/^.*version ([0-9]+\.[0-9]+\.[0-9]+)/\1/p' |
+          awk '{print "tag_version=v"$1}' >> "$GITHUB_ENV"
+      - name: Release version
+        run: echo "${{ env.tag_version }}"
+        # TODO - we may find it necessary to override this check at some point in order to "hotfix" a version, but it
+        #  seems prudent even in that scenario to simply release a new version
+      - name: Check release version doesn't yet exist
+        run: |
+          if [ $(git tag -l '${{ env.tag_version }}') ]; then echo "Release already found - aborting" && exit 1; else echo "Continuing with release"; fi
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{ env.tag_version }}
+          generate_release_notes: true
+      - name: Run latest-tag
+        uses: EndBug/latest-tag@latest
```

Comment:
Summarizing this -

- Check out the full repo history (necessary for checking tags that may already exist),
- Install python / library deps,
- Create an env var named `tag_version` that is simply `v{__version__}`,
- Verify that we haven't already created a tag for this version
  -  If we have, the build will fail and the version will need to be incremented
- Create release `v{__version__}`
- Tag this release as `latest`